src/format/LemLibOdomGeneratorV0_5.tsx

```diff
@@ -0,0 +1,381 @@
+  @ValidateNumber(num => num > 0 && num <= 1000) // Don't use IsEnum
+  @Expose()
+  uol: UnitOfLength = UnitOfLength.Inch;
+  @IsPositive()
+  @Expose()
+  pointDensity: number = 1; // inches
+  @IsPositive()
+  @Expose()
+  controlMagnetDistance: number = 5 / 2.54;
+  @Type(() => FieldImageSignatureAndOrigin)
+  @ValidateNested()
+  @IsObject()
+  @Expose()
+  fieldImage: FieldImageSignatureAndOrigin<FieldImageOriginType> =
+    getDefaultBuiltInFieldImage().getSignatureAndOrigin();
+  @IsString()
+  @MinLength(1)
+  @Expose()
+  lemlibMethod: LemLibMethod = LemLibMethod.MoveToPoint;
+  @Expose()
+  chassisName: string = "chassis";
+  @ValidateNumber(num => num >= 0)
+  @Expose()
+  movementTimeout: number = 5000;
+  @IsBoolean()
+  @Expose()
+  setPose: boolean = true;
+  @Exclude()
+  private format_: LemLibOdomGeneratorFormatV0_5;
+
+  constructor(format: LemLibOdomGeneratorFormatV0_5) {
+    this.format_ = format;
+    makeAutoObservable(this);
+
+    initGeneralConfig(this);
+  }
+
+  get format() {
+    return this.format_;
+  }
+
+  getConfigPanel() {
+    return <GeneralConfigPanel config={this} />;
+  }
+}
+
+// observable class
+class PathConfigImpl implements PathConfig {
+  @Exclude()
+  speedLimit: EditableNumberRange = {
+    minLimit: { value: 0, label: "0" },
+    maxLimit: { value: 600, label: "600" },
+    step: 1,
+    from: 40,
+    to: 120
+  };
+  @Exclude()
+  bentRateApplicableRange: EditableNumberRange = {
+    minLimit: { value: 0, label: "0" },
+    maxLimit: { value: 1, label: "1" },
+    step: 0.001,
+    from: 0,
+    to: 0.1
+  };
+  @Exclude()
+  bentRateApplicationDirection = BentRateApplicationDirection.HighToLow;
+  @Exclude()
+  maxDecelerationRate: number = 127;
+
+  @Exclude()
+  readonly format: LemLibOdomGeneratorFormatV0_5;
+
+  @Exclude()
+  public path!: Path;
+
+  constructor(format: LemLibOdomGeneratorFormatV0_5) {
+    this.format = format;
+    makeAutoObservable(this);
+  }
+
+  getConfigPanel() {
+    return (
+      <>
+        <Typography>(No setting)</Typography>
+      </>
+    );
+  }
+}
+
+// observable class
+export class LemLibOdomGeneratorFormatV0_5 implements Format {
+  isInit: boolean = false;
+  uid: string;
+
+  private gc = new GeneralConfigImpl(this);
+  private readonly disposers: (() => void)[] = [];
+
+  constructor() {
+    this.uid = makeId(10);
+    makeAutoObservable(this);
+  }
+
+  createNewInstance(): Format {
+    return new LemLibOdomGeneratorFormatV0_5();
+  }
+
+  getName(): string {
+    return "LemLib Odom Code Gen v0.5.x (inch)";
+  }
+
+  register(app: MainApp): void {
+    if (this.isInit) return;
+    this.isInit = true;
+
+    this.disposers.push(
+      app.history.addEventListener("beforeExecution", event => {
+        if (event.isCommandInstanceOf(AddCubicSegment)) {
+          // UX: Cancel the event and replace the cubic segment with a linear segment
+          event.isCancelled = true;
+          let targetPath: Path | undefined = app.interestedPath();
```

Comment:
I understand what you are trying to do, good idea. But you should be able to replace line 273 to line 288 to the following code:
```tsx

          const cancelledCommand = event.command;
          app.history.execute(
            `Add linear segment with end control point ${cancelledCommand.end.uid} to path ${cancelledCommand.path.uid}`,
            new AddLinearSegment(cancelledCommand.path, cancelledCommand.end)
          );
```